cmd/ci-secret-bootstrap/main.go

```diff
@@ -0,0 +1,182 @@
+	fs.StringVar(&o.configPath, "config", "", "Path to the config file to use for this tool.")
+	fs.StringVar(&o.bwUser, "bw-user", "", "Username to access BitWarden.")
+	fs.StringVar(&o.bwPasswordPath, "bw-password-path", "", "Path to a password file to access BitWarden.")
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Errorf("cannot parse args: '%s'", os.Args[1:])
+	}
+	return o
+}
+
+func (o *options) validateOptions() error {
+	if o.bwUser == "" {
+		return fmt.Errorf("the mandatory arg bw-user is empty")
+	}
+	if o.bwPasswordPath == "" {
+		return fmt.Errorf("the mandatory arg bw-password-path is empty")
+	}
+	if o.configPath == "" {
+		return fmt.Errorf("the mandatory arg config is empty")
+	}
+	return nil
+}
+
+func (o *options) completeOptions() error {
+	bytes, err := ioutil.ReadFile(o.bwPasswordPath)
+	if err != nil {
+		return err
+	}
+	o.bwPassword = strings.TrimSpace(string(bytes))
+
+	kubeConfigs, _, err := util.LoadKubeConfigs(o.kubeConfigPath)
+	if err != nil {
+		return err
+	}
+	o.kubeConfigs = kubeConfigs
+
+	config, err := loadConfig(o.configPath)
+	if err != nil {
+		return err
+	}
+	o.config = config
+
+	return validateConfig(kubeConfigs, config)
+}
+
+type Config struct {
+	Secrets []SecretConfig `json:"secrets"`
+}
+
+type BWContext struct {
+	BWItem     string `json:"bwItem"`
+	Field      string `json:"field"`
+	Attachment string `json:"attachment"`
+}
+
+type SecretContext struct {
+	Cluster   string `json:"cluster"`
+	Namespace string `json:"namespace"`
+	Name      string `json:"name"`
+}
+
+type SecretConfig struct {
+	From map[string]BWContext `json:"from"`
+	To   []SecretContext      `json:"to"`
+}
+
+func loadConfig(path string) (*Config, error) {
+	bytes, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	c := &Config{}
+	err = yaml.Unmarshal(bytes, c)
+	if err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+func validateConfig(kConfigs map[string]rest.Config, c *Config) error {
+	//TODO
+	// case1: either field or attachment but not both
+	// case2: cluster is a context in the kubeconfig
+	return nil
+}
+
+func main() {
+	o := parseOptions()
+	if err := o.validateOptions(); err != nil {
+		logrus.WithError(err).Fatal("Invalid arguments.")
+	}
+	if err := o.completeOptions(); err != nil {
+		logrus.WithError(err).Fatal("Failed to complete options.")
+	}
+	bwClient := bitwarden.NewBitwardenClient()
+	if err := bwClient.Login(o.bwUser, o.bwPassword); err != nil {
+		logrus.WithError(err).Fatal("Failed to login Bitwarden.")
+	}
+	defer func() {
+		if err := bwClient.Logout(); err != nil {
+			logrus.WithError(err).Fatal("Failed to login Bitwarden.")
+		}
+	}()
+
+	for _, secretConfig := range o.config.Secrets {
+		data := make(map[string][]byte)
+		for key, bwContext := range secretConfig.From {
+			if bwContext.Field != "" {
+				fieldValue, err := bwClient.GetField(bwContext.BWItem, bwContext.Field)
```

Comment:
You can make this all less verbose by only getting the value in the conditional and do the error handling, conversion and assignment just once:

```
var value string
var err error
if field {
  value, err = GetField()
} else {
  value, err = GetAttachment()
}
if err != nil {
  error out
}
data[key] = []byte(value)
```

Reply:
Done.
Thought about this too and chose the other way.